sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -88,16 +88,27 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
         s"Selected $selected partitions out of $total, pruned $percentPruned% partitions."
       }
 
+      // need to add projections from combineFilters in
+      val combineFilter = combineFilters.reduceLeftOption(expressions.And)
```

Comment:
existing: `combine` is kind of a confusing name.  How about `partitionAndNormalColumnFilters`.  Its really long, but this code is pretty tough to follow.  I'd rather error on readability.